### ISSUE TITLE
Visualization in MatchZoo?

### ISSUE THREAD

**Issue by mingmingyang1** [faneshion/MatchZoo#44]
how to achieve visualization in zoo ? For example, how to draw the loss curve

**Comment by uduse** [comment]
I don't think it's supported now.

**Comment by yangliuy** [comment]
You can draw the training loss curve by parsing the output log file. It should be quite straightforward with Pyplot.

**Comment by mingmingyang1** [comment]
How to do it ?  Can you explain it in detail?
Thank you very much!

**Comment by yangliuy** [comment]
You can refer to my code as follows:
`'''
Parse the log files of MatchZoo to draw the learning curves of 
deep learning model training process to observe the training loss
and metrics on train/dev/test data. With these curves, we can do 
a better job in model debugging
For ranking with hinge loss
@author: Liu Yang (yangliuyx@gmail.com / lyang@cs.umass.edu)
'''
import os
import sys
from matplotlib import rcParams

def draw_train_learning_curve(log_file, log_label):
    command = 'curl ' + log_file 
    r = os.popen(command) 
    info = r.readlines()  
    start_line = '[Model] Model Compile Done.'
    start_flag = False
    train_loss = []
    valid_map = []
    valid_p1 = []
    test_map = []
    test_p1 = []
    for line in info: 
        line = line.strip('\r\n')
        if start_flag:
            # print line
            tokens = line.split('\t')
            # print 'tokens: ', tokens
            if 'train' in line:
                #print tokens
                train_loss.append(float(tokens[2].split('=')[1]))
            elif 'valid' in line:
                #print tokens
                if len(tokens) < 6:
                    continue
                valid_map.append(float(tokens[2].split('=')[1]))
                valid_p1.append(float(tokens[5].split('=')[1]))
            else:
                #print tokens[2], tokens[5]
                test_map.append(float(tokens[2].split('=')[1]))
                test_p1.append(float(tokens[5].split('=')[1]))
        if start_line in line.strip('\r\n'):
            start_flag = True

    # draw learning curve
    import matplotlib.pyplot as plt

    SMALL_SIZE = 8
    MEDIUM_SIZE = 10
    BIGGER_SIZE = 14

    plt.rc('font', size=SMALL_SIZE)          # controls default text sizes
    plt.rc('axes', titlesize=SMALL_SIZE)     # fontsize of the axes title
    plt.rc('axes', labelsize=MEDIUM_SIZE)    # fontsize of the x and y labels
    plt.rc('xtick', labelsize=SMALL_SIZE)    # fontsize of the tick labels
    plt.rc('ytick', labelsize=SMALL_SIZE)    # fontsize of the tick labels
    plt.rc('legend', fontsize=SMALL_SIZE)    # legend fontsize
    plt.rc('figure', titlesize=BIGGER_SIZE)  # fontsize of the figure title

    x = range(len(train_loss))
    #print 'x', x
    #print train_loss
    if (len(train_loss) != len(test_p1)) or (len(train_loss) != len(valid_p1)):
        print 'wait a few seconds for the valid/test metrics in the next iteration...'
        min_len = min(len(test_p1), len(valid_p1))
        train_loss = train_loss[0:min_len]
        valid_map = valid_map[0:min_len]
        valid_p1 = valid_p1[0:min_len]
        test_map = test_map[0:min_len]
        test_p1 = test_p1[0:min_len]
        x = x[0:min_len]
    line1, = plt.plot(x, train_loss, 'r-v', label='train_loss')
    line2, = plt.plot(x, valid_map, 'c-s', label='valid_map')
    line3, = plt.plot(x, valid_p1, 'm-o', label='valid_p1')
    line4, = plt.plot(x, test_map, 'b-+', label='test_map')
    line5, = plt.plot(x, test_p1, 'g-^', label='test_p1')
    plt.legend(handles=[line1, line2, line3, line4, line5], loc=1)
    rcParams['grid.linestyle'] = 'dotted'
    plt.grid()
    plt.title(log_label)
    plt.ylabel('model training learning curves')
    plt.show()

exp_id = 1
print 'Exp ', exp_id, ': model comparation'
for data in list(['data']):
    for model in list(['arci', 'arcii', 'matchpyramid']): 
        log_file = 'path_of_log_file'
        log_label = data +' ' + model + ' training'
        draw_train_learning_curve(log_file, log_label)

**Comment by bwanglzu** [comment]
@mingmingyang1 Thanks for your question, I'll close it for now, if you have further questions please feel free to re-open it or create a new issue.
